x-pack/test/api_integration/deployment_agnostic/default_configs/serverless.config.base.ts

```diff
@@ -125,6 +125,8 @@ export function createServerlessTestConfig<T extends DeploymentAgnosticCommonSer
           ...svlSharedConfig.get('kbnTestServer.serverArgs'),
           ...kbnServerArgsFromController[options.serverlessProject],
           `--serverless=${options.serverlessProject}`,
+          // Enable custom roles
+          '--xpack.security.roleManagementEnabled=true',
```

Comment:
I believe the right place to make a change is ES server argument:
https://github.com/elastic/kibana/blob/bba42298efb8bcc103c4c437323fb4a449be6432/x-pack/test/api_integration/deployment_agnostic/default_configs/serverless.config.base.ts#L115-L118

For Kibana you need to update `config/serverless.oblt.yml` and remove `xpack.security.roleManagementEnabled: false`